### ISSUE TITLE
3.6.0.CR1  do not support http2? 

### ISSUE THREAD

**Issue by heyile** [eclipse-vertx/vert.x#2716]
the same code run correctly in the 3.5.3
when I use 3.6.0.CR1, it throws 

Nov 19, 2018 3:25:35 PM io.vertx.core.http.impl.HttpClientRequestImpl
SEVERE: javax.net.ssl.SSLHandshakeException: Failed to create SSL connection

**Comment by heyile** [comment]
https://github.com/heyile/vertx-demo/tree/h2Support


    <dependency>
      <groupId>io.vertx</groupId>
      <artifactId>vertx-core</artifactId>
      <version>3.6.0.CR1</version>
    </dependency>

    <dependency>
      <groupId>io.vertx</groupId>
      <artifactId>vertx-core</artifactId>
      <version>3.5.3</version>
    </dependency>

**Comment by vietj** [comment]
I will have a look

**Comment by vietj** [comment]
did you update boring SSL to use 2.0.12.Final instead of 2.0.7.Final ?

**Comment by vietj** [comment]
it works for me actually with 3.6.0.CR1

**Comment by vietj** [comment]
for now labelled as question as I've not been able to reproduce it

**Comment by BillyYccc** [comment]
I think this might be related to SSL connection handshake, cause I'm meeting exceptions when running [tests](https://github.com/eclipse-vertx/vert.x/blob/master/src/test/java/io/vertx/core/net/NetTest.java) of vertx-core.
If I'm using openjdk-8 or openjdk-11, everything is fine, but when I switch to Oracle JDK 8, the tests do not work.

#### System Environment
> ➜  ~ uname -r
4.19.2-300.fc29.x86_64
➜  vert.x git:(master) java -version
java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)

```
Starting test: NetTest#testNetClientInternalTLS 
javax.net.ssl.SSLHandshakeException: Failed to create SSL connection
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:329)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:315)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:307)
	at io.netty.handler.ssl.SslUtils.handleHandshakeFailure(SslUtils.java:318)
	at io.netty.handler.ssl.SslHandler.setHandshakeFailure(SslHandler.java:1551)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1023)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.channels.ClosedChannelException
	at io.netty.handler.ssl.SslHandler.channelInactive(...)(Unknown Source)
java.lang.AssertionError: Failed to create SSL connection
	at org.junit.Assert.fail(Assert.java:88)
	at io.vertx.test.core.AsyncTestBase.fail(AsyncTestBase.java:262)
	at io.vertx.test.core.AsyncTestBase.lambda$onSuccess$2(AsyncTestBase.java:640)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connect$2(NetClientImpl.java:116)
	at io.vertx.core.net.impl.NetClientImpl.doFailed(NetClientImpl.java:239)
	at io.vertx.core.net.impl.NetClientImpl.lambda$failed$10(NetClientImpl.java:235)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:180)
	at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:235)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$5(NetClientImpl.java:202)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$1(ChannelProvider.java:81)
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:113)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:329)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:315)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:307)
	at io.netty.handler.ssl.SslUtils.handleHandshakeFailure(SslUtils.java:318)
	at io.netty.handler.ssl.SslHandler.setHandshakeFailure(SslHandler.java:1551)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1023)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Unhandled exception 
java.lang.AssertionError: Failed to create SSL connection
	at org.junit.Assert.fail(Assert.java:88)
	at io.vertx.test.core.AsyncTestBase.fail(AsyncTestBase.java:262)
	at io.vertx.test.core.AsyncTestBase.lambda$onSuccess$2(AsyncTestBase.java:640)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connect$2(NetClientImpl.java:116)
	at io.vertx.core.net.impl.NetClientImpl.doFailed(NetClientImpl.java:239)
	at io.vertx.core.net.impl.NetClientImpl.lambda$failed$10(NetClientImpl.java:235)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:180)
	at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:235)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$5(NetClientImpl.java:202)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$1(ChannelProvider.java:81)
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:113)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:329)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:315)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:307)
	at io.netty.handler.ssl.SslUtils.handleHandshakeFailure(SslUtils.java:318)
	at io.netty.handler.ssl.SslHandler.setHandshakeFailure(SslHandler.java:1551)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1023)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)

```

```
Starting test: NetTest#testHostVerificationHttpsMatching 
Client from origin /127.0.0.1:50968 failed to connect over ssl: java.lang.NoSuchFieldError: state 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.core.net.NetTest.lambda$null$205(NetTest.java:2667)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connect$2(NetClientImpl.java:116)
	at io.vertx.core.net.impl.NetClientImpl.doFailed(NetClientImpl.java:239)
	at io.vertx.core.net.impl.NetClientImpl.lambda$failed$10(NetClientImpl.java:235)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:180)
	at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:235)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$5(NetClientImpl.java:202)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$1(ChannelProvider.java:81)
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:113)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:329)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:315)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:307)
	at io.netty.handler.ssl.SslUtils.handleHandshakeFailure(SslUtils.java:318)
	at io.netty.handler.ssl.SslHandler.setHandshakeFailure(SslHandler.java:1551)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1023)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Unhandled exception 
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.vertx.test.core.AsyncTestBase.assertTrue(AsyncTestBase.java:372)
	at io.vertx.core.net.NetTest.lambda$null$205(NetTest.java:2667)
	at io.vertx.core.net.impl.NetClientImpl.lambda$connect$2(NetClientImpl.java:116)
	at io.vertx.core.net.impl.NetClientImpl.doFailed(NetClientImpl.java:239)
	at io.vertx.core.net.impl.NetClientImpl.lambda$failed$10(NetClientImpl.java:235)
	at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:320)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:188)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:180)
	at io.vertx.core.net.impl.NetClientImpl.failed(NetClientImpl.java:235)
	at io.vertx.core.net.impl.NetClientImpl.lambda$doConnect$5(NetClientImpl.java:202)
	at io.vertx.core.net.impl.ChannelProvider.lambda$connect$1(ChannelProvider.java:81)
	at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:113)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:329)
	at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:315)
	at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:307)
	at io.netty.handler.ssl.SslUtils.handleHandshakeFailure(SslUtils.java:318)
	at io.netty.handler.ssl.SslHandler.setHandshakeFailure(SslHandler.java:1551)
	at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1023)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1429)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:947)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:822)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

and several other failure tests not listed here.

**Comment by vietj** [comment]
what is your OS ?

**Comment by vietj** [comment]
what is your JDK version ?

**Comment by vietj** [comment]
try upgrading jetty ALPN version to 2.0.9

**Comment by vietj** [comment]
it seems it's required for latest Oracle JDK

**Comment by heyile** [comment]
I will have a try with ssl 2.0.12 tomorrow.  the jdk which I use is Oracle1.8. maybe it's related with jdk

**Comment by heyile** [comment]
my os is ubuntu1.6.4

**Comment by heyile** [comment]
@vietj  thx, I resolved it with boring ssl 2.0.12
